lib/commands/serve.js

```diff
@@ -11,6 +11,8 @@ const clearModule = require('clear-module')
 
 const { createError } = micro
 
+const lamdbaMap = new Map()
```

Comment:
I created a map which will cache imported lambdas upon usage. This was necessary to fix a problem I encountered when requiring a lambda that was importing and using firebase.

Reply:
Can you provide me with the full firebase related error? We can't really allow for module cache reusing for this would break the concept of self-enclosed, no-shared-context of lambda executions. I tried something with child processes, but it didn't work so far. Maybe we can clear a broader module scope on `clearModule.match(...)` to ensure _anything_ needed by the lambda gets cleared out and re-loaded...

Reply:
This was the error output of the second time you'd request a lambda using firebase:

```
Error: Module did not self-register.
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:718:18)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/[REDACTED]/node_modules/grpc/src/grpc_extension.js:32:13)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
```

Not really helpful. My only quick thought was that the module being required multiple times was breaking some firebase internal module relationships.

Maybe yes, a broader clean up of modules is the key.

Reply:
@Meesayen do you mind to test your firebase related code using branch [fs/broader-cache-cleanup](https://github.com/lucasconstantino/now-we-test/tree/fs/broader-cache-cleanup)?

Reply:
Sorry, I couldn't have a look at this lately.
Here's what I'll do, I'll get rid of the caching part on this PR, and unblock the basic typescript support. The firebase issue can wait (firebase doesn't work when deployed on Now anyway...)